Source/JavaScriptCore/parser/Parser.h

```diff
@@ -1289,19 +1293,24 @@ class Parser {
 
     ScopeRef pushScope()
     {
+        ImplementationVisibility implementationVisibility = m_implementationVisibility;
         LexicalScopeFeatures lexicalScopeFeatures = NoLexicalFeatures;
         bool isFunction = false;
         bool isGenerator = false;
         bool isArrowFunction = false;
         bool isAsyncFunction = false;
         if (!m_scopeStack.isEmpty()) {
+            implementationVisibility = m_scopeStack.last().implementationVisibility();
+            if (m_scopeStack.last().isFunctionBoundary() && implementationVisibility != ImplementationVisibility::PrivateRecursive)
+                implementationVisibility = ImplementationVisibility::Public;
```

Comment:
If we do this (scopes inside a function have Public while the function has Private), do we need to have `m_scopeStack.last().isFunctionBoundary() && ` part?

Reply:
ah hmm yeah good point

we need the `m_scopeStack.last().isFunctionBoundary()` part because the `Parser` adds its own top-level `Scope` and we need to make sure that the `ImplementationVisibility` propagates at least one level down

i want to move this code inside `setIsFunction()` but that doesn't have access to the `m_scopeStack`

lemme noodle on this a bit more 🤔